metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -326,15 +325,19 @@ def _configure_transforms(self) -> None:
         # Add the system metadata transformer at the end of the list.
         self.transformers.append(SystemMetadataTransformer(self.ctx))
 
-    def _configure_reporting(
-        self, report_to: Optional[str], no_default_report: bool
-    ) -> None:
-        if report_to == "datahub":
+    def _configure_reporting(self, report_to: Optional[str]) -> None:
+        if self.dry_run:
+            # In dry run mode, we don't want to report anything.
+            return
+
+        if not report_to:
+            # Reporting is disabled.
+            pass
+        elif report_to == "datahub":
```

Comment:
We should move this string to const at some point

Reply:
yup - can do that in a follow up